Batch processing refactor

meffmadd opened the pull request:
Could not reproduce #55 but refactored and added tests

meffmadd committed:
First creation of test cases

meffmadd committed:
Refactoring batch processing loop

meffmadd committed:
More major refactorings

meffmadd committed:
Bring back old async queue
Use logging

meffmadd committed:
Make batch test go brrr

meffmadd committed:
Added test to test large responses

meffmadd committed:
Merge branch 'main' into 55-responses-are-dropped-from-outputerror-files-in-the-batch-api

# Conflicts:
#	aqueduct/management/models.py

natkam reviewed (commented):
I feel like I need more background about what's happening here! Can we talk about it on Monday?

meffmadd committed:
Change is_duplicate method into _get_processing_id method and simply check for existence in if-clause
Change methods into static methods where applicable

meffmadd committed:
Add enum to batch models and use it in batches.py

meffmadd committed:
Fix incorrect batch env variable name for AQUEDUCT_BATCH_PROCESSING_MIN_CONCURRENCY
Document use for batch_processing_concurrency function in settings.py via docstring

natkam reviewed (commented):
A few nit-picky comments because I couldn't find anything really serious that could have caused the bug. Overall it seems to me that the code is more readable after the refactoring!

meffmadd committed:
Updated exceptions for finer grained errors
Made db requests in batch_cancel more efficient
Changed semaphore to BoundedSemaphore
Removed unused cancelled attribute in AsyncBoundedParallelQueue
